### DISCUSSION TITLE
Unable to add tuya smart plugs

### DISCUSSION THREAD

**Discussion by ChriZathens** [make-all/tuya-local#1926] [Q&A]
Hello guys! I would appreciate your help!
I acquired 3 new smart plugs which work with tuya and they operate as expected using the smart life app.
I tried to get the local keys for those to be able to add them to home assistant, but I get an error that my subscription has ended.
So I used tinytuya and managed to get all device IDs and keys.
However when I enter the ID of a device, the secret and the IP and having auto for the protocol, I always get "Unable to connect to your device with those details. It could be an intermittent issue, or they may be incorrect."
I also have one dehumidifier already configured since before my tuya trial account expired. If I go at that device and click on configure I see the local key and the IP and if I click submit I get a "Success" message.
I see the already registered key there and the one that I obtained using tinytuya and they are the same, so there is no reason to believe that the local keys for the 3 plugs are wrong..
Any ideas please?
Thanks in advance!!!

**Comment by make-all** [comment]
`python -m tinytuya scan` should report the protocol, "auto" does not always work.

**Comment by AdrianGarside** [comment]
I added a bunch of smart plugs recently and they were a lot more finickety about responding to the queries. But they come back as protocol 3.1 FWIW
